### DISCUSSION TITLE
Make handleBlock Methods that Bypass Island Recalculation

### DISCUSSION THREAD

**Discussion by ErwanDev** [BG-Software-LLC/SuperiorSkyblock2#738] [Denied Feature Requests] [answered]
These are current methods within the Island API:

   void handleBlockPlace(Key key, int amount);
 
  void handleBlockBreak(Key key, int amount);

  It would be incredibly helpful if you created a version of these methods that functioned as is but were not recalculated during island recalculation. These methods are great for making sure the worth of custom (plugin-based) blocks are registered for worth. The only issue with using this method directly is that when island recalculation is run, any custom block that was sent to Superior using this method is dropped (worth for these blocks are reset to 0). Since anyone using these methods would handle the breaking of their custom blocks (and then call handleBlockBreak), it would be logical to skip these blocks so that their worth is still included after recalculation. The developer would then be able to remove the worth of the block if their block is destroyed using the provided API methods above.

   Alternatively (and probably a better solution), is to add a set Worth method within the IslandWorthCalculatedEvent event. There is currently a getWorth function but no ability to modify the new worth value after its calculated. This would be a great API feature and would help developers greatly.
   
```
      public IslandWorthCalculatedEvent(Island island, SuperiorPlayer player, BigDecimal level, BigDecimal worth) {
        super(island);
        this.player = player;
        this.level = level;
        this.worth = worth;
    }

    /**
     * Get the player who requested the operation.
     * Can be null if the console called the operation.
     */
    @Nullable
    public SuperiorPlayer getPlayer() {
        return player;
    }

    /**
     * Get the new level of the island.
     */
    public BigDecimal getLevel() {
        return level;
    }

    /**
     * Get the new worth value of the island.
     */
    public BigDecimal getWorth() {
        return worth;
    }
    
    /**
    * Set the new worth value of the island.
     */
    public void setWorth(BigDecimal worth) { 
        this.worth = worth;
    }

}
    
```

**Comment by OmerBenGera** [comment, marked as answer]
Use the built-in system to handle custom values for your blocks. Register a custom block-key for your blocks and give them a worth in the worth-file using the custom block keys. An example can be found on our wiki page.

**Reply by ErwanDev** [reply]
I still believe that a setWorth method within the IslandWorthCalculatedEvent would be highly beneficial for other purposes.

**Reply by OmerBenGera** [reply]
> I still believe that a setWorth method within the IslandWorthCalculatedEvent would be highly beneficial for other purposes.

It cannot be done due to how the system works.
Also, by doing so you break the idea of the system - its idea is to calculate worth of all the blocks. Therefore, if I let you change the worth value, breaking blocks can lead to negative numbers.
